qcodes/instrument_drivers/Keysight/keysight_b220x.py

```diff
@@ -0,0 +1,181 @@
+from qcodes import VisaInstrument
+from qcodes.utils.validators import MultiType, Ints, Enum, Lists
+
+import re
+
+
+class KeysightB220X(VisaInstrument):
+    """
+    QCodes driver for B2200 / B2201 switch matrix
+    """
+
+    def __init__(self, name, address, **kwargs):
+        super().__init__(name, address, terminator='\r', **kwargs)
+
+        self.add_function(name='clear_status',
+                          call_cmd='*CLS',
+                          docstring='Clears status register and error queue of the instrument.')
+
+        self.add_function(name='reset',
+                          call_cmd='*RST',
+                          docstring='Performs an instrument reset. Does not reset error queue!')
+
+        self.add_parameter(name='get_status',
+                           get_cmd='*ESR?',
+                           get_parser=lambda status_str: int(status_str),
+                           docstring='Queries status register.')
+
+        self.add_parameter(name='get_error',
+                           get_cmd=':SYST:ERR?',
+                           docstring='Queries error queue')
+
+        self.add_parameter(name='connections',
+                           get_cmd=':CLOS:CARD? 0',
+                           get_parser=KeysightB220X.parse_channel_list,
+                           docstring='queries currently active connections and returns a set of tuples {(input, output), ...}'
+                           )
+
+        self.add_function(name='connect',
+                          call_cmd=self._connect,
+                          args=[Ints(1, 14), Ints(1, 48)],
+                          docstring='connects given pair. Two arguments: input, output')
+
+        self.add_function(name='disconnect',
+                          call_cmd=self._disconnect,
+                          args=[Ints(1, 14), Ints(1, 48)],
```

Comment:
I tempt to recommend defining `Ints(1, 14)` as `self._available_input_ports` (or similar) and `Ints(1, 48)` as `self._available_output_ports` so that these definitions are re-used.

Reply:
Good point. I will do that.